### PR TITLE
Guest test executor refactoring

### PR DESCRIPTION
**Pull request by hongyuni**

### CODE REVIEW

**Comment by hongyuni** [comment]
Hi @ysun , would you help on this guest-test code architecture refactoring review, it's been agreed based on Rex Zhang's new guest-test feature adding proposal, hopefully we could start with new features from more contributors like Rex start from this new refactoring done.

File: guest-test/guest.test_executor.sh [hunk 1]
```diff
@@ -8,13 +8,20 @@
...
 
 # @desc This script prepare and run $TESTCASE in Guest VM
+# based on $FEATURE selection
 
 ###################### Variables ######################
-SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
-echo "$SCRIPT_DIR"
+# exec only if script being executed
+if [ "${BASH_SOURCE[0]}" -ef "$0" ]; then
+  SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
```
Comment (ysun) [inline guest-test/guest.test_executor.sh:16]: Hi, are the quotation marks really necessary? Or is it for coding style checking? 
Why not `SCRIPT_DIR=$( cd $( dirname "$0" ) && pwd )`   ?

File: guest-test/tdx_osv_sanity/tdx_osv_sanity.test_executor.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# History:  5, Dec., 2023 - Hongyu Ning - creation
+
+
+# @desc This script prepare and run $TESTCASE in for $FEATURE tdx_osv_sanity
+# in tdx VM; please note, $FEATURE name and subfolder name under guest-test
+# must be the exactly same, here FEATURE=tdx_osv_sanity
+
+###################### Variables ######################
+## common variables example ##
+SCRIPT_DIR_LOCAL="$( cd "$( dirname "$0" )" && pwd )"
```
Comment (ysun) [inline guest-test/tdx_osv_sanity/tdx_osv_sanity.test_executor.sh:16]: The same question as the previous one about the quotation marks.

File: guest-test/tdx_osv_sanity/tdx_osv_sanity.test_executor.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+# get test scenario config for $FEATURE tdx test_executor
+source "$SCRIPT_DIR_LOCAL"/../test_params.py
+## end of common variables example ##
+
+###################### Functions ######################
+## $FEATURE specific Functions ##
+
+###################### Do Works ######################
+## common works example ## 
+cd "$(dirname "$0")" 2>/dev/null || exit 1
```
Comment (ysun) [inline guest-test/tdx_osv_sanity/tdx_osv_sanity.test_executor.sh:27]: That's my concern about the quote marks, what do you really want to quote? Here are "$(dirname " and ")". Are they correct?

**Comment by ysun** [comment]
Thank Hongyu for the quick response!

**Pull request merged**